Firmware/Marlin_main.cpp

```diff
@@ -1581,32 +1636,33 @@ void setup()
     fw_crash_init();
 
 #ifdef UVLO_SUPPORT
-  if (eeprom_read_byte((uint8_t*)EEPROM_UVLO) != PowerPanic::NO_PENDING_RECOVERY) { //previous print was terminated by UVLO
+  if (printer_recovering()) { //previous print was terminated by UVLO
       manage_heater(); // Update temperatures 
-#ifdef DEBUG_UVLO_AUTOMATIC_RECOVER 
-		printf_P(_N("Power panic detected!\nCurrent bed temp:%d\nSaved bed temp:%d\n"), (int)degBed(), eeprom_read_byte((uint8_t*)EEPROM_UVLO_TARGET_BED));
-#endif
-     uvlo_auto_recovery_ready = (degBed() > ( (float)eeprom_read_byte((uint8_t*)EEPROM_UVLO_TARGET_BED) - AUTOMATIC_UVLO_BED_TEMP_OFFSET));
-     if (uvlo_auto_recovery_ready){
-          #ifdef DEBUG_UVLO_AUTOMATIC_RECOVER 
+      //Restore printing type
+      saved_printing_type = eeprom_read_byte((uint8_t*)EEPROM_UVLO_PRINT_TYPE);
+#ifdef DEBUG_UVLO_AUTOMATIC_RECOVER
```

Comment:
Personally I really don't like this `DEBUG_UVLO_AUTOMATIC_RECOVER` code. It makes the code very unreadable. We can probably move all the serial print messages into `recover_print()` function.

Just an observation/comment. Not a blocker.

Reply:
It is 6 years old and as DEBUG code has been placed close to the operation you are trying to debug.
In VScode and an existing `Configuration_prusa.h` it is "grayed out" and so not that disturbing to read at least for me.